tests/smoke/helm_test.go

```diff
@@ -0,0 +1,222 @@
+
+	err = hs.installer.CreateK8sRookOperatorViaHelm()
+	require.NoError(hs.T(), err)
+
+	require.True(hs.T(), kh.IsPodInExpectedState("rook-operator", "rook", "Running"),
+		"Make sure rook-operator is in running state")
+
+	time.Sleep(10 * time.Second)
+
+	err = hs.installer.CreateK8sRookCluster()
+	require.NoError(hs.T(), err)
+
+	err = hs.installer.CreateK8sRookToolbox()
+	require.NoError(hs.T(), err)
+
+	hs.helper, err = clients.CreateTestClient(enums.Kubernetes, kh)
+	require.Nil(hs.T(), err)
+
+	hs.pvcDef = `apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: {{.claimName}}
+  annotations:
+    volume.beta.kubernetes.io/storage-class: rook-block
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1M`
+
+	hs.storageclassDef = `apiVersion: rook.io/v1alpha1
+kind: Pool
+metadata:
+  name: {{.poolName}}
+  namespace: rook
+spec:
+  replication:
+    size: 1
+  # For an erasure-coded pool, comment out the replication count above and uncomment the following settings.
+  # Make sure you have enough OSDs to support the replica count or erasure code chunks.
+  #erasureCode:
+  #  codingChunks: 2
+  #  dataChunks: 2
+---
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+   name: rook-block
+provisioner: rook.io/block
+parameters:
+    pool: {{.poolName}}`
+}
+
+func (hs *HelmSmokeSuite) TearDownSuite() {
+	if hs.T().Failed() {
+		hs.k8sh.GetRookLogs("rook-operator", "default", hs.T().Name())
+		hs.k8sh.GetRookLogs("rook-api", "rook", hs.T().Name())
+		hs.k8sh.GetRookLogs("rook-ceph-mgr", "rook", hs.T().Name())
+		hs.k8sh.GetRookLogs("rook-ceph-mon", "rook", hs.T().Name())
+		hs.k8sh.GetRookLogs("rook-ceph-osd", "rook", hs.T().Name())
+		hs.k8sh.GetRookLogs("rook-ceph-rgw", "rook", hs.T().Name())
+		hs.k8sh.GetRookLogs("rook-ceph-mds", "rook", hs.T().Name())
+	}
+	hs.installer.UninstallRookFromK8sViaHelm()
+
+}
+
+//Test to make sure all rook components are installed and Running
+func (hs *HelmSmokeSuite) TestRookInstallViaHelm() {
+
+	assert.True(hs.T(), hs.k8sh.CheckPodCountAndState("rook-operator", "default", 1, "Running"),
+		"Make sure there is 1 rook-operator present in Running state")
+	assert.True(hs.T(), hs.k8sh.CheckPodCountAndState("rook-api", "rook", 1, "Running"),
+		"Make sure there is 1 rook-api present in Running state")
+	assert.True(hs.T(), hs.k8sh.CheckPodCountAndState("rook-ceph-mgr", "rook", 1, "Running"),
+		"Make sure there is 1 rook-ceph-mgr present in Running state")
+	assert.True(hs.T(), hs.k8sh.CheckPodCountAndState("rook-ceph-osd", "rook", 1, "Running"),
+		"Make sure there is at lest 1 rook-ceph-osd present in Running state")
+	assert.True(hs.T(), hs.k8sh.CheckPodCountAndState("rook-ceph-mon", "rook", 3, "Running"),
+		"Make sure there are 3 rook-ceph-mon present in Running state")
+}
+
+//Test BlockCreation on Rook that was installed via Helm
+func (hs *HelmSmokeSuite) TestBlockStoreOnRookInstalledViaHelm() {
```

Comment:
is there a way we can re-use the block test code from the main smoke test? I would only expect to see test code here for starting the operator via helm. All the block test code could be shared with the block test. No need to rewrite the block test here. A lot of the setup validation could be shared as well.